implementations/rust/ockam/ockam_app/src/projects/mod.rs

```diff
@@ -1,8 +1,24 @@
-use ockam_api::cloud::project::Project;
+use std::sync::Arc;
+use tauri::async_runtime::RwLock;
+
+use ockam_api::cloud::{project::Project, share::RoleInShare};
 
 pub(crate) mod commands;
 mod error;
 pub(crate) mod events;
 pub(super) mod plugin;
 
 pub(crate) type State = Vec<Project>;
+pub(crate) type SyncState = Arc<RwLock<State>>;
+
+pub(crate) trait ProjectExt {
```

Comment:
I would put this code alongside the `Project` code since it is entirely the same domain of project / users / roles.

Reply:
Makes sense, I moved it, currently visible in 2bdd06efa13cfe8fbb1271527b9ab2b5a5d62338.